### ISSUE TITLE
How to use the focal loss while training my classification model

### ISSUE THREAD

**Issue by spaul13** [AlexeyAB/darknet#5283]
I want to add focal loss for my dataset training for classification to tackle the class imbalance problem. As I went through the focal_loss only applicable for yolo layer.
_Can anyone please tell me how to add a yolo layer at the end of any classification model?_
![image](https://user-images.githubusercontent.com/20827182/79952940-27986800-8449-11ea-9688-7e633c3e9335.png)

In my custom cfg file, I replace the last layer from 
**[softmax]
groups=1**
to
**[yolo]
focal_loss = 1**

before the last layer, I have a convolution filter (1X1) where # of filters = # of output labels or classes

that gives me the error shown in the picture above. Can anyone please tell me how to get rid of the error?

**Comment by AlexeyAB** [comment]
No, you can't use focal_loss for training Classification model.

**Comment by spaul13** [comment]
@AlexeyAB can u please suggest me any other techniques other than using focal_loss for classification model to tackle the class imbalance problem?

what I already tried is that I have 100 classes and the max class has 64 images so I repeat same files in .list multiple times to make the # of images under each class to be 64 (similar to oversampling) but for me with so many classes and diverse # of images under each class, it's not working well as expected.

Can u please tell me any other techniques for the classification model?
